src/main/java/com/gregtechceu/gtceu/api/mui/theme/ThemeAPI.java

```
@@ -0,0 +1,154 @@
+    protected final Object2ObjectMap<String, WidgetThemeParser> widgetThemeFunctions = new Object2ObjectOpenHashMap<>();
+    protected final Object2ObjectOpenHashMap<String, String> jsonScreenThemes = new Object2ObjectOpenHashMap<>();
+    private final Object2ObjectMap<String, String> screenThemes = new Object2ObjectOpenHashMap<>();
+
+    private ThemeAPI() {
+        registerWidgetTheme(Theme.PANEL,
+                new WidgetTheme(GuiTextures.MC_BACKGROUND, null, Color.WHITE.main, 0xFF404040, false),
+                WidgetTheme::new);
+        registerWidgetTheme(Theme.BUTTON, new WidgetTheme(GuiTextures.MC_BUTTON, GuiTextures.MC_BUTTON_HOVERED,
+                Color.WHITE.main, Color.WHITE.main, true), WidgetTheme::new);
+        registerWidgetTheme(Theme.ITEM_SLOT,
+                new WidgetSlotTheme(GuiTextures.SLOT_ITEM, Color.withAlpha(Color.WHITE.main, 0x60)),
+                WidgetSlotTheme::new);
+        registerWidgetTheme(Theme.FLUID_SLOT,
+                new WidgetSlotTheme(GuiTextures.SLOT_FLUID, Color.withAlpha(Color.WHITE.main, 0x60)),
+                WidgetSlotTheme::new);
+        registerWidgetTheme(Theme.TEXT_FIELD, new WidgetTextFieldTheme(0xFF2F72A8, 0xFF5F5F5F),
+                (parent, json, fallback) -> new WidgetTextFieldTheme(parent, fallback, json));
+        registerWidgetTheme(Theme.TOGGLE_BUTTON,
+                new WidgetThemeSelectable(GuiTextures.MC_BUTTON, GuiTextures.MC_BUTTON_HOVERED, Color.WHITE.main,
+                        Color.WHITE.main, true,
+                        GuiTextures.MC_BUTTON_DISABLED, IDrawable.NONE, Color.WHITE.main, Color.WHITE.main, true),
+                WidgetThemeSelectable::new);
+    }
+
+    @Override
+    public ITheme getDefaultTheme() {
+        return DEFAULT_DEFAULT;
+    }
+
+    @Override
+    public @NotNull ITheme getTheme(String id) {
+        return this.themes.getOrDefault(id, getDefaultTheme());
+    }
```

Do we want some sort of log or warning incase a theme is fetched but not found instead of silently returning default?